Fix Image block field to wrap the entire image across left and right alignments.

MostafaMagdyy opened the pull request:
Fixes #5922 for the image field container to wrap the entire image in right and left alignment.


For full-width alignment, I think it's another issue itself. Disregard the field to wrap the entire image because the image is going outside the whole container and above the sidebar on the right if you try to drag its block. So, I do not think this PR should fix it, because it is not related to the failure of the wrapper to fill the whole image. Let me know if I should fix it in this PR, and I will do so. Also, please clarify how the full width should be, **because I do not see any difference between it and when the image is aligned center with size=large, as both take size=100vw in getImageBlockSizes function**.

MostafaMagdyy commented:
Please review this pull request. @stevepiercy ,@polyester, could you please take a look and provide your feedback? Thank you!

stevepiercy commented:
@MostafaMagdyy the lint CI check fails. You must fix it before review will happen. See https://6.docs.plone.org/volto/contributing/linting.html

stevepiercy commented on packages/volto/news/5922.bugfix:
```diff
@@ -0,0 +1 @@
+Fix image block field to wrap the entire image across various alignments.@MostafaMagdyy
```
```suggestion
Fix image block field to wrap the entire image across various alignments. @MostafaMagdyy
```

stevepiercy reviewed (changes requested):
Just missing a space in the change log.

MostafaMagdyy commented:
@stevepiercy Thank you for your feedback. I have addressed all your comments and fixed all the issues. Additionally, here are the screenshots displaying the new behavior for left and right alignment.
![oie_PHzGyA7xqc6d](https://github.com/plone/volto/assets/97239596/29e26164-a8ee-4b6f-917c-fa597619f895)
![oie_cLX8eZVwphge](https://github.com/plone/volto/assets/97239596/70be776a-0a8f-46ee-aea1-248312313bfc)

stevepiercy reviewed (approved):
I approve, but this still needs review from a Volto Team member.

ichim-david reviewed (changes requested):
@MostafaMagdyy you don't fully understand why the select area with the border around the image is as small as it is.
It's because of a float property.
That is useful when you want to have text to the right or left of your image.
Here is a demo on the current feature in action:
![before-flex](https://github.com/plone/volto/assets/152852/cf44c4ad-bea2-419b-9938-b371cff18ecf)

and how it would function with your changes:
![after-flex](https://github.com/plone/volto/assets/152852/3741c86d-e013-474e-9676-ad3457f286a9)

You need some css that will affect the editing selected area alongside
extra code to have the text float to the left or right.

MostafaMagdyy commented:
> @MostafaMagdyy you don't fully understand why the select area with the border around the image is as small as it is. It's because of a float property. That is useful when you want to have text to the right or left of your image. Here is a demo on the current feature in action: ![before-flex](https://private-user-images.githubusercontent.com/152852/318426333-cf44c4ad-bea2-419b-9938-b371cff18ecf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE5NzcwMTgsIm5iZiI6MTcxMTk3NjcxOCwicGF0aCI6Ii8xNTI4NTIvMzE4NDI2MzMzLWNmNDRjNGFkLWJlYTItNDE5Yi05OTM4LWIzNzFjZmYxOGVjZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwMVQxMzA1MThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iNDc4MjE4MTQyZjQ1ZDZiZjAwMWI1NDIxMjU2YjFjODlmZGEyNWZhNzIyZjgwYjk5YWZlOGNjMWQzZTQzNjllJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.wavHYzyVrWzig914p5XVg1YfLY2yAx5CfEWqkV4BRlA)
> 
> and how it would function with your changes: ![after-flex](https://private-user-images.githubusercontent.com/152852/318426400-3741c86d-e013-474e-9676-ad3457f286a9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE5NzcwMTgsIm5iZiI6MTcxMTk3NjcxOCwicGF0aCI6Ii8xNTI4NTIvMzE4NDI2NDAwLTM3NDFjODZkLWUwMTMtNDc0ZS05Njc2LWFkMzQ1N2YyODZhOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwMVQxMzA1MThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05NDA4ZmNmYjZmNTQ1ZTQ2OThhYTUwMDhjYzc5MWM5Y2QyODhhMWE3ODRlYTZkNjYzMGZlYzg2NzYyNTM0NjE5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.wwKlDvAThrI9o-kuwMg0sKcuUm1J2TEQN6OhuAavmMI)
> 
> You need some css that will affect the editing selected area alongside extra code to have the text float to the left or right.

Thank you for your feedback. 
Now i understand the importance of it and i will try to make the same behaviour as it was.

davisagli commented:
It looks like this PR did not solve the problem in a way that supported floating images next to text, and has been abandoned, so I will close it.
